File: src/main/java/org/usfirst/frc/team6325/robot/commands/Drive/ArcadeJoystickDrive.java
package org.usfirst.frc.team6325.robot.commands.Drive;

import edu.wpi.first.wpilibj.command.Command;

import org.usfirst.frc.team6325.robot.Robot;


public class ArcadeJoystickDrive extends Command 
{
	public ArcadeJoystickDrive() 
	{
		// Use requires() here to declare subsystem dependencies
		requires(Robot.drivetrain);
	}

	// Called repeatedly when this Command is scheduled to run
	@Override
	protected void execute() 
	{
		double left = Robot.oi.drivetrainLeft;
		double right = Robot.oi.drivetrainRight;
		Robot.drivetrain.drive (left, right);
	}

	// Make this return true when this Command no longer needs to run execute()
	@Override
	protected boolean isFinished() 
	{
		return false;
	}

	// Called once after isFinished returns true
	@Override
	protected void end() 
	{
		Robot.drivetrain.killMotors();
	}

	// Called when another command which requires one or more of the same
	// subsystems is scheduled to run
	@Override
	protected void interrupted() 
	{
		Robot.drivetrain.killMotors();
	}
}

File: src/main/java/org/usfirst/frc/team6325/robot/commands/Drive/ArcadeJoystickDrive.kt
package org.usfirst.frc.team6325.robot.commands.Drive

import org.sertain.command.*
import org.usfirst.frc.team6325.robot.Polybius


class ArcadeJoystickDrive : Command()
{
    init { requires Polybius.drivetrain }

    override fun execute()
    {
        val sliderModifier = 1 - ((Polybius.oi.sliderRawAxisNumber + 1)/2)
        System.err.println("SliderModifer is " + sliderModifier);
		val left = Polybius.oi.drivetrainLeft * sliderModifier;
		val right = Polybius.oi.drivetrainRight * sliderModifier;
        Polybius.drivetrain.drive(left, right);
    }

    override fun isCanceled() = Polybius.drivetrain.killMotors()
    override fun onDestroy() = Polybius.drivetrain.killMotors()
}